Source/UnrealGT/Private/Triggers/GTTimedGeneratorTrigger.cpp

```diff
@@ -14,27 +14,14 @@ UGTTimedGeneratorTrigger::UGTTimedGeneratorTrigger()
 {
 }
 
-void UGTTimedGeneratorTrigger::Trigger()
-{
-    FDateTime TimeStamp = FDateTime::Now();
-    // TODO re-add trigger all option
-    // TODO so linkedimage generators in actor info are triggered before actor
-    // ifno segmentation
-    for (const FGTGeneratorReference& GeneratorReference : DataGenerators)
-    {
-        UGTDataGeneratorComponent* GeneratorComponent = GeneratorReference.GetComponent(GetOwner());
-        if (GeneratorComponent)
-        {
-            GeneratorComponent->GenerateData(TimeStamp);
-        }
-    }
-}
-
 // Called when the game starts
 void UGTTimedGeneratorTrigger::BeginPlay()
 {
     Super::BeginPlay();
 
-    GetWorld()->GetTimerManager().SetTimer(
-        TriggerTimerHandle, this, &UGTTimedGeneratorTrigger::Trigger, 1 / FrameRate, true);
-}
+    if (FrameRate != 0.0)
+    {
+        GetWorld()->GetTimerManager().SetTimer(
+            TriggerTimerHandle, this, &UGTTimedGeneratorTrigger::Trigger, 1 / FrameRate, true);
+    }
```

Comment:
Instead of silently failing, this should either:

- Use UIMin/UIMax metadata specifiers to disallow valuers equal or less 0
- or Trigger every frame (see suggestion)
```suggestion
    const auto TimerInterval = FrameRate == 0.F ? 0.F : 1 / FrameRate;
    GetWorld()->GetTimerManager().SetTimer(
            TriggerTimerHandle, this, &UGTTimedGeneratorTrigger::Trigger, TimerInterval, true);
```

Reply:
I'd argue that a user might set the framerate to zero when (s)he wants the module to trigger zero times per second (i.e. never fire). I actually tried to do this when I was testing a different change, thinking I could use it as a temporary "disable". So I wasn't actually thinking of this as a silent fail, but rather implementing the behavior that the user has requested.

That said, I'm happy to leave this however you'd like me to. The change that I care about is the refactor, this was just some boyscouting along the way. Let me know which solution you want to go with.

Reply:
You are right, zero meaning never trigger makes more sense than zero meaning always trigger. I am fine with it "failing"/not setting the timer. I think it would also be good to extend this behaviour to negative frame rates.
```suggestion
    if (FrameRate <= 0.F)
    {
        GetWorld()->GetTimerManager().SetTimer(
            TriggerTimerHandle, this, &UGTTimedGeneratorTrigger::Trigger, 1 / FrameRate, true);
    }
```